Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -548,12 +568,46 @@ void SetLidOpen()
     GenerateDIInterrupt(INT_CVRINT);
 }
 
-bool ChangePartition(u64 offset)
+bool UpdateRunningGameMetadata(u64 title_id)
 {
+  if (!s_inserted_volume)
+    return false;
+
   DVDThread::WaitUntilIdle();
-  const bool success = s_inserted_volume->ChangePartition(offset);
-  FileMonitor::SetFileSystem(s_inserted_volume.get());
-  return success;
+
+  const DiscIO::Partition& partition =
+      s_inserted_volume->GetVolumeType() == DiscIO::Platform::WII_DISC ? s_current_partition :
+                                                                         DiscIO::PARTITION_NONE;
+
+  u64 volume_title_id;
+  if (!s_inserted_volume->GetTitleID(&volume_title_id, partition))
+    return false;
+
+  if (volume_title_id != title_id)
+    return false;
+
+  SConfig::GetInstance().SetRunningGameMetadata(*s_inserted_volume, partition);
+  return true;
+}
+
+bool UpdateRunningGameMetadata()
```

Comment:
I suppose they could be combined if we had a nice way to indicate that there is no `title_id` parameter. I wonder when we're going to get `std::optional` :)